psydac/api/fem.py

```diff
@@ -187,24 +187,38 @@ def __init__(self, expr, kernel_expr, *args, **kwargs):
         # ...
         if len(domain)>1:
             i,j = self.get_space_indices_from_target(domain, target )
-            test_space  = self.spaces[0].spaces[i]
-            trial_space = self.spaces[1].spaces[j]
+            trial_space  = self.spaces[0].spaces[j]
+            test_space = self.spaces[1].spaces[i]
         else:
             trial_space  = self.spaces[0]
             test_space   = self.spaces[1]
 
+        # ...
+        test_ext  = None
+        trial_ext = None
         if isinstance(target, Boundary):
-            axis      = target.axis
-            test_ext  = target.ext
-            trial_ext = target.ext
+            axis        = target.axis
+            ext         = target.ext
+            test_ext    = ext
+            trial_ext   = ext
+            test_grid   = QuadratureGrid( test_space, axis, test_ext)
+            trial_grid  = QuadratureGrid( trial_space, axis, trial_ext)
+            self._grid  = (test_grid,)
         elif isinstance(target, Interface):
-            axis       = target.axis
-            test_ext   = -1 if isinstance(self.kernel_expr.test,  PlusInterfaceOperator) else 1
-            trial_ext  = -1 if isinstance(self.kernel_expr.trial, PlusInterfaceOperator) else 1
+            axis         = target.axis
+            test         = self.kernel_expr.test
+            trial        = self.kernel_expr.trial
+            test_target  = target.plus if isinstance(test, PlusInterfaceOperator) else target.minus
+            trial_target = target.plus if isinstance(trial, PlusInterfaceOperator) else target.minus
+            test_ext     = test_target.ext
+            trial_ext    = trial_target.ext
+            test_grid    = QuadratureGrid( test_space, axis, test_ext)
+            trial_grid   = QuadratureGrid( trial_space, axis, trial_ext)
+            self._grid   = (test_grid, trial_grid) if test_target==target.minus else (trial_grid, test_grid)
```

Comment:
It is not clear to me why test and trial grid are swapped in the case when `test_target==target.minus`. Can you please explain?

Reply:
when we have an integral over an interface we can have two cases:
 -  integral(v_minus*u_plus) where v_minus is the test function in the minus patch and u_plus is the trial function in the plus patch
 - integral(v_plus*u_minus) which is the other way around
 and the condition above tells us in which case we are

Reply:
I see. Do we ever have the other two cases, i.e.

- `integral(v_minus * u_minus)`
- `integral(v_plus * u_plus)`

which would contribute to the diagonal blocks of a bilinear form?

Reply:
`integral(v_minus * u_minus)` is converted to a boundary integral in the minus domain in Sympde, the same goes for the other case, so Psydac doesn't see it as an interface integral

Reply:
I see. I had not realized that some interface integrals were converted back into boundary integrals of the single patches.

To briefly recap, the [`Boundary` case](https://github.com/pyccel/psydac/blob/a2521bc528fd1edff4f81d7d7487e25f57dccf26/psydac/api/fem.py#L199-L205) will always assemble contributions to the diagonal blocks, while the [`Interface` case](https://github.com/pyccel/psydac/blob/a2521bc528fd1edff4f81d7d7487e25f57dccf26/psydac/api/fem.py#L206-L216) will always assemble contributions to the off-diagonal blocks. Do I get it right?

I think this needs to be properly explained somewhere. @campospinto what do you think?

Reply:
yes, that's correct

Reply:
Yes, this is an important ingredient of the assembly for mapped domains and it should be clearly explained. I would say that `psydac/api/fem.py` is a natural place to explain this discretization procedure.

Reply:
It's Sympde that does the transformation, Psydac will give the wrong results if the transformation was not done, so it should be in Sympde

Reply:
ok, then I suggest to describe the transformation in SymPDE, and also add some words in Psydac since the discrete quadrature assumes this particular transformation.

Reply:
We should explain this in both Psydac and SymPDE. In this PR we need to explain what Psydac expects from SymPDE, because Psydac will not work properly if the transformation is not done properly.

We should also add some checks to catch possible errors, like `integral(u_plus * v_plus)` on an `Interface`... Do you think this is doable?

Reply:
yes, this should be doable

Reply:
where should we add the explanations?

Reply:
done in [137d7f1](https://github.com/pyccel/psydac/pull/160/commits/137d7f143b72d4578a47bf313c05516e50dfbe12)